backend/app/analysis/proximity.py

```diff
@@ -0,0 +1,87 @@
+import nltk
+from collections import Counter
+from more_itertools import windowed
+
+from ..models import (
+    Document,
+    Gender,
+    Corpus,
+)
+
+
+def run_analysis(corpus_id, word_window):
+    """
+    Generates a dictionary of dictionaries for each `Document` object. Each dictionary maps a `Gender` to a word count
+    of words within a specified window of that `Gender`'s pronouns.
+
+    :param corpus_id: An int representing a `Corpus` instance
+    :param word_window: An integer describing the number of words to look at of each side of a gendered word
+
+    :return: A dict mapping `Document` ids to a dict mapping strings (`Gender` labels) to a `Counter` instance.
+    """
+    results = {}
+    genders = set(Gender.objects.all())
```

Comment:
Ultimately, we'll want the user to be able to specify which genders they want to see as part of this analysis. For now, though, I think grabbing `.all` is fine.